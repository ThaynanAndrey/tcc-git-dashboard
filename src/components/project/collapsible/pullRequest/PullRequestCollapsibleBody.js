import React from 'react';
import ProjectPullRequests from '../../../pull_request/ProjectPullRequests';

/**
 * Stateless component with the function of presenting a Collapsible Body to Pull Requests.
 * 
 * @author Thaynan Nunes
 */
const PullRequestCollapsibleBody = ({ idProject, openPullRequest }) => (
    <div className="collapsible-body body-collapsible-project">
        <ProjectPullRequests idProject={idProject} openPullRequest={openPullRequest} />
    </div> 
);

export default PullRequestCollapsibleBody;