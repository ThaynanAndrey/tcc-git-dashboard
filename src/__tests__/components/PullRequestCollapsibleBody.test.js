import React from 'react';
import { shallow, configure } from "enzyme";
import toJson from 'enzyme-to-json';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import PullRequestCollapsibleBody from '../../components/project/collapsible/pullRequest/PullRequestCollapsibleBody';

describe('Test RepositoriesNoProject', () => {
  it('renders correctly a component to show PullRequestCollapsibleBody', () => {
    const openPullRequest = jest.fn();
    const idProject = '123';

    const tree = shallow(
      <PullRequestCollapsibleBody 
        idProject={idProject}
        openPullRequest={openPullRequest}
      />
    );

    expect(toJson(tree)).toMatchSnapshot();
  });

  
  it('the component should have classes collapsible-body body-collapsible-project', () => {
    const tree = shallow(
      <PullRequestCollapsibleBody />
    );

    expect(tree.find(".collapsible-body.body-collapsible-project").exists()).toBeTruthy();
  });
})