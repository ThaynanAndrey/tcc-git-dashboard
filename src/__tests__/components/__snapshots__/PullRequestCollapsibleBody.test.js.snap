// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RepositoriesNoProject renders correctly a component to show PullRequestCollapsibleBody 1`] = `
<div
  className="collapsible-body body-collapsible-project"
>
  <AuthenticatedComponent
    idProject="123"
    openPullRequest={[MockFunction]}
  />
</div>
`;
